components/create/Collection.vue

```diff
@@ -0,0 +1,349 @@
+                rel="nofollow noopener noreferrer">
+                Learn more
+              </a>
+            </p>
+          </div>
+        </div>
+      </NeoField>
+    </form>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type {
+  CollectionToMintBasilisk,
+  CollectionToMintKusama,
+  CollectionToMintStatmine,
+} from '@/composables/transaction/types'
+import type { PalletBalancesAccountData } from '@polkadot/types/lookup'
+
+import {
+  NeoButton,
+  NeoField,
+  NeoIcon,
+  NeoInput,
+  NeoSelect,
+  NeoSwitch,
+} from '@kodadot1/brick'
+import DropUpload from '@/components/shared/DropUpload.vue'
+import { availablePrefixes, depositAmount } from '@/utils/chain'
+import { Interaction } from '@kodadot1/minimark/v1'
+import { notificationTypes, showNotification } from '@/utils/notification'
+import {
+  getAssetIdByAccount,
+  getAssetMetadataById,
+  getKusamaAssetId,
+} from '@/utils/api/bsx/query'
+import { balanceOf } from '@kodadot1/sub-api'
+import { CHAINS, Prefix } from '@kodadot1/static'
+import { decodeAddress, encodeAddress } from '@polkadot/util-crypto'
+import format from '@/utils/format/balance'
+
+// composables
+const { accountId } = useAuth()
+const { apiInstanceByPrefix } = useApi()
+const { transaction, status, isLoading } = useTransaction()
+const { urlPrefix, setUrlPrefix, assets } = usePrefix()
+
+// form state
+const logo = ref<File | null>(null)
+const name = ref('')
+const description = ref('')
+const unlimited = ref(true)
+const max = ref(1)
+const symbol = ref('')
+
+// balance state
+const balance = ref()
+const collectionDeposit = ref()
+const canDeposit = computed(() => {
+  return parseFloat(balance.value) >= parseFloat(collectionDeposit.value)
+})
+
+const depositTokenId = ref()
+const depositSymbol = ref()
+const depositAsset = computed(() => assets(depositTokenId.value))
+const depositAssetSymbol = computed(
+  () => depositSymbol.value || depositAsset.value.symbol
+)
+
+const menus = availablePrefixes().filter(
+  (menu) => menu.value !== 'movr' && menu.value !== 'glmr'
+)
+const chainByPrefix = menus.find((menu) => menu.value === urlPrefix.value)
+const selectBlockchain = ref(chainByPrefix?.value || menus[0].value)
+
+const currentChain = computed(() => {
+  return selectBlockchain.value as string
+})
+
+watchEffect(() => setUrlPrefix(currentChain.value as Prefix))
+
+const isKusama = computed(
+  () => currentChain.value === 'ksm' || currentChain.value === 'rmrk'
+)
+
+const isBasilisk = computed(
+  () => currentChain.value === 'bsx' || currentChain.value === 'snek'
+)
+
+const isAssetHub = computed(
+  () => currentChain.value === 'ahk' || currentChain.value === 'ahp'
+)
+
+const createCollection = async () => {
+  let collection:
+    | CollectionToMintBasilisk
+    | CollectionToMintKusama
+    | CollectionToMintStatmine = {
+    file: logo.value,
+    name: name.value,
+    description: description.value,
+    nftCount: unlimited.value ? 0 : max.value,
```

Comment:
`CollectionToMintBasilisk` doesn't have nftCount

could you use `BaseCollectionType`?

Reply:
updated